lib/user.rb

```diff
@@ -0,0 +1,37 @@
+require_relative "db_connection"
+require 'bcrypt'
+
+class User
+  attr_reader :email, :id, :name
+  def self.create(name, email, pass)
+    # magic encyption
+    encrypted_pass = BCrypt::Password.create(pass)
+
+    add_user = DbConnection.query("INSERT INTO users (username, email, pass) VALUES ('#{name}','#{email}', '#{encrypted_pass}');")
+  end
+
+  def self.find(email)
+    find_user = DbConnection.query("SELECT * FROM users WHERE email = '#{email}';")
+    User.new(email: find_user[0]["email"], id: find_user[0]["id"], name: find_user[0]["username"]) 
+  end
+
+  def self.find_by_id(id)
+    return nil unless !id.nil?
+    find_user = DbConnection.query("SELECT * FROM users WHERE id = #{id};")
```

Comment:
Might want to check the DB was able to retrieve a record for this `id` before serializing into your model